### PR TITLE
Landing actions

### PR DESCRIPTION
**Pull request by andrewjmeier**
I just connected up the landing actions for properties with the game loop that I had

### CODE REVIEW

**Comment by andrewjmeier** [comment]
@konnorbeard want to take a quick look over this? I had to change the way the super methods were being called. Instead of super.method.apply(this) I used call(this, player...) I think this way you can pass parameters to the super class.

File: src/board/properties/housingProperty.js [hunk 1]
```diff
@@ -1,19 +1,62 @@
...
 
-HousingProperty.prototype.performLandingAction = function(player) { 
-  // todo
-  // see above railroad property fn comments
+HousingProperty.prototype.performLandingAction = function(game) {
+
+    HousingProperty.super_.prototype.performLandingAction.call(this, game);
+};
+
+HousingProperty.prototype.getRent = function(game) {
```
Comment (konnorbeard) [inline src/board/properties/housingProperty.js]: nice

File: src/board/properties/housingProperty.js [hunk 2]
```diff
@@ -1,19 +1,62 @@
...
+};
+
+HousingProperty.prototype.isMonopoly = function() {
+  var groupCount = 0;
+  for (i in this.owner.properties) {
+    if (this.owner.properties[i].propertyGroup === this.propertyGroup) {
+      groupCount++;
+    }
+  }
+  if (this.propertyGroup === PG_BROWN || this.propertyGroup === PG_BLUE) {
```
Comment (konnorbeard) [inline src/board/properties/housingProperty.js]: could shorten this to: 

`return (this.propertyGroup === PG_BROWN || this.propertyGroup === PG__BLUE) ? groupCount === 2 : groupCount === 3`

File: src/board/properties/property.js [hunk 3]
```diff
@@ -1,20 +1,33 @@
...
-}; 
+Property.prototype.performLandingAction = function(game) {
+  var player = game.getCurrentPlayer();
+  if (this.owner === player) return;
+
+  if (this.owner === null) {
+    if (player.money > this.cost) {
+      player.makePayment(this.cost);
+      player.properties.push(this);
+      this.owner = player;
```
Comment (konnorbeard) [inline src/board/properties/property.js]: I feel like a buyProperty(player) function here might be useful later

File: src/board/properties/railroadProperty.js [hunk 4]
```diff
@@ -1,20 +1,38 @@
...
+
+RailroadProperty.prototype.getRent = function(game) {
+    var rrCount = 0;
+    for (i in this.owner.properties) {
+        if (this.owner.properties[i].propertyGroup === this.propertyGroup) {
+            rrCount++;
+        }
+    }
+
+    return this.rent[rrCount - 1];
```
Comment (konnorbeard) [inline src/board/properties/railroadProperty.js]: if this.owner is null then we could have this.rent[-1]. might want some safety checks here even if we only plan on calling it when there's an owner

File: src/board/taxes/tax.js [hunk 5]
```diff
@@ -1,17 +1,19 @@
...
 
-Tax.prototype.performLandingAction = function(player) { 
+Tax.prototype.performLandingAction = function(game) {
   // should probably refactor this into a fn
   // that checks balance, prompts player to mortage/sell
   // before declaring him bankrupt
-  this.player.makePayment(this.amount);
-}; 
+  game.getCurrentPlayer().makePayment(this.taxAmount);
+  Tax.super_.prototype.performLandingAction.call(this, game);
```
Comment (konnorbeard) [inline src/board/taxes/tax.js]: Thanks for adding all these super_ calls, was on my todo but hadn't gotten around to it

File: src/cards/chanceDeck.js [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+        var player = game.getCurrentPlayer();
+        if (player.position > 11 && player.position < 28) {
+            // move to water works
+            player.moveTo(28);
+        } else {
+            // move to electric company
+            player.moveTo(12);
+        }
+
+        // TODO - pay owner 10x dice
```
Comment (konnorbeard) [inline src/cards/chanceDeck.js]: this will work nicely with getRent()*2

File: src/cards/chanceDeck.js [hunk 7]
```diff
@@ -0,0 +1,126 @@
...
+
+    var chance11 = new Card("Make general repairs on all your property - for each house pay $25 - for each hotel $100", function(game) {
+        var player = game.getCurrentPlayer();
+        var housesCount = 0;
+        var hotelsCount = 0;
+        for (var prop in player.properties) {
+            if (player.properties.hasOwnProperty(prop)) {
+            // or if (Object.prototype.hasOwnProperty.call(obj,prop)) for safety...
+                // TODO - link up correctly w/ property object
+                housesCount += player.properties[prop].numberOfHouses;
```
Comment (konnorbeard) [inline src/cards/chanceDeck.js]: should be numHouses (or need to change the var names in housingproperty.js)

File: src/cards/chanceDeck.js [hunk 8]
```diff
@@ -0,0 +1,126 @@
...
+                if (player.properties[prop].hasHotel) {
```
Comment (konnorbeard) [inline src/cards/chanceDeck.js]: we could make this a method ie 

`Property.prototype.hasHotel = function() { 
    return this.numHouses == 5;
}`

otherwise need to just check for 5 houses here

File: src/cards/chanceDeck.js [hunk 9]
```diff
@@ -0,0 +1,126 @@
...
+    });
+
+    var chance11 = new Card("Make general repairs on all your property - for each house pay $25 - for each hotel $100", function(game) {
+        var player = game.getCurrentPlayer();
+        var housesCount = 0;
+        var hotelsCount = 0;
+        for (var prop in player.properties) {
+            if (player.properties.hasOwnProperty(prop)) {
+            // or if (Object.prototype.hasOwnProperty.call(obj,prop)) for safety...
+                // TODO - link up correctly w/ property object
```
Comment (konnorbeard) [inline src/cards/chanceDeck.js]: just saw this todo comment, maybe you already knew the following two comments i made

File: src/cards/communityChestDeck.js [hunk 10]
```diff
@@ -0,0 +1,102 @@
...
+
+    var card11 = new Card("Pay School Fees of $50", function(game) {
+        game.getCurrentPlayer().makePayment(50);
+    });
+
+    var card12 = new Card("Receive $25 Consultancy Fee", function(game) {
+        game.getCurrentPlayer().makeDeposit(25);
+    });
+
+    var card13 = new Card("You are assessed for street repairs - $40 per house, $115 per hotel", function(game) {
```
Comment (konnorbeard) [inline src/cards/communityChestDeck.js]: repeated code here but i think it might confuse things more to abstract this to card or something so maybe it's ok

File: src/game.js [hunk 11]
```diff
@@ -50,18 +72,20 @@ Game.prototype.move = function() {
...
     }
-    nextPosition = this.players[this.currentPlayer].position + spacesToMove;
-    if (nextPosition >= 40) {
-        nextPosition = nextPosition % 40;
-        this.players[this.currentPlayer].money += 200;
-    }
-    this.players[this.currentPlayer].moveTo(nextPosition);
+    this.getCurrentPlayer().move(spacesToMove);
+    var player = this.getCurrentPlayer();
+    this.board.spaces[player.position].performLandingAction(this
```
Comment (konnorbeard) [inline src/game.js]: weird paren newline here

File: src/player.js [hunk 12]
```diff
@@ -38,11 +59,24 @@ Player.prototype.makeDeposit = function(amount) {
 };
 
 Player.prototype.moveTo = function(position) {
```
Comment (konnorbeard) [inline src/player.js]: should this have 'collectGo' as an argument? (ie do not pass go, do not collection 200..)

**Comment by konnorbeard** [comment]
@andrewjmeier lgtm just see comments. From what I understand apply() and call() do the same thing (except i guess call allows arguments) so that's fine w/ me. nice work, i like what you did here.
